### PR TITLE
Refactored hook handlers in mod_csi

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_csi` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150052) / small_tests / bb62a1ba261a7672742d685ac84bc65c170d9378
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3846/150052/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3846/150052/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-08_10.29.23/lib.mongooseim.logs/run.2022-11-08_10.29.24/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/150147) / small_tests / 6562a08257eb067115604b75d2c0aae3456cca64
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3846/150147/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3846/150147/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-08_11.26.06/lib.mongooseim.logs/run.2022-11-08_11.26.06/suite.log.html)

File: src/mod_csi.erl [hunk 1]
```diff
@@ -63,10 +63,12 @@ supported_features() ->
...
 %%%
 
--spec c2s_stream_features([exml:element()], mongooseim:host_type(), jid:lserver()) ->
-          [exml:element()].
-c2s_stream_features(Acc, _HostType, _Lserver) ->
-    lists:keystore(<<"csi">>, #xmlel.name, Acc, csi()).
+-spec c2s_stream_features(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: [exml:element()],
+    Params :: map(),
+    Extra :: map().
```
Comment (NelsonVides) [inline src/mod_csi.erl:69]: I think that we can use now in all these handlers the new type, `gen_hook:extra()`, just for good habit, but nevertheless it doesn't change much in this PR, just for the future 👌🏽

**Review by NelsonVides — APPROVED** [review]
👍🏽

**Pull request merged**